src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1436,11 +1436,14 @@ abstract class RefChecks extends Transform {
           checkTypeRefBounds(ann.tpe, tree)
         }
 
-        annots
-          .map(_.transformArgs(transformTrees))
-          .groupBy(_.symbol)
```

Comment:
FTR `groupBy` was the culprit here as it doesn't preserve the original iteration order.

Reply:
hm, So call `groupBy` on a `List` Will not gives me a `ListMap`?

Reply:
Nope, it gives back a `HashMap` currently: https://github.com/scala/scala/blob/2d9e6acce8c4246e6cba5a22ff9d965ec3bd117c/src/library/scala/collection/Iterable.scala#L538-L568

Reply:
I just looked into the code too, I think that should change. The `groupBy` method on `LinearSeq` should return a `LinkedHashMap`